### PR TITLE
Update windows-capture and make it compile on windows

### PR DESCRIPTION
**Pull request by NiiightmareXD**
I think we need a better structure to make cross-platform development easier for example we need to implement traits for each os but for now I made it to compile on windows.

windows-capture updates:
[Add Crop 🎉](https://github.com/NiiightmareXD/windows-capture/commit/35c31f5e5581fddd6ccf5bd33d01eb42068d6830)
[Better Threading 🔥](https://github.com/NiiightmareXD/windows-capture/commit/edccad0402cfce9510147552b7a2170f49299015)
[Bug Fix ⭐](https://github.com/NiiightmareXD/windows-capture/commit/a76944c7a66c23a80cda2629c07a5e3f023428b2)
[Performance Boost 🔥](https://github.com/NiiightmareXD/windows-capture/commit/90b324aeeae8e5f532fc6a0ed54cd367fcb568c9)
[Bug Fix 🐛](https://github.com/NiiightmareXD/windows-capture/commit/73cecbe191309ac11b4030c5fb8a260bbb933a6a)
[Optimization 🔥](https://github.com/NiiightmareXD/windows-capture/commit/97021e5078988d496ee516bf4930f7ae62915dcf)
[Better Error Message 😊](https://github.com/NiiightmareXD/windows-capture/commit/6df8228df70213e72706d1a9a18699f245b02868)
[Update Monitor Struct ⭐](https://github.com/NiiightmareXD/windows-capture/commit/62b3eb082183758b90e6ce272a9518bad3c3bbad)
[Mutable Buffer 🔥](https://github.com/NiiightmareXD/windows-capture/commit/df9c1d4b683c667e0abfc1646167708f5ac52c73)
[Bug Fix 🐛](https://github.com/NiiightmareXD/windows-capture/commit/cd77276d5ff8986081730a1146b30d2a56645526)
[Low Latency 🔥](https://github.com/NiiightmareXD/windows-capture/commit/230b17382131f07ee49728e3889cd6a64549fca0)
[Optimization 🔥](https://github.com/NiiightmareXD/windows-capture/commit/bace569eac82f4f374c7d9cb75a0c0c3e78d2b5c)
[Update thiserror ✅](https://github.com/NiiightmareXD/windows-capture/commit/9c24f6be887fc8d53e7dbe05e911bdfc76459aa6)
[Native Windows Encoder 🔥](https://github.com/NiiightmareXD/windows-capture/commit/692f63bf5933979dfb5ea07fb1e0f790b5143bfc)

About FFmpeg: the latest version of the crate used for FFmpeg doesn't compile on windows I added a specific commit for windows that compiles but it shouldn't be a problem because I'm going to stabilize the native windows encoder and drop ffmpeg for windows completely its already used for `save_as_image`.

### CODE REVIEW

File: src/frame/mod.rs [hunk 1]
```diff
@@ -1,35 +1,34 @@
...
-    access::ScreenCaptureAccess,
-    display::{CGDirectDisplayID, CGDisplay, CFArrayGetCount, CFArrayGetValueAtIndex, CFArrayRef, },
+    sc_sys::{
+        CMSampleBufferGetImageBuffer, CMSampleBufferGetPresentationTimeStamp,
+        CMSampleBufferGetSampleAttachmentsArray, SCFrameStatus,
+    },
 };
-use core_video_sys::{CVPixelBufferRef, CVPixelBufferLockBaseAddress, CVPixelBufferGetWidth, CVPixelBufferGetHeight, CVPixelBufferGetBaseAddressOfPlane, CVPixelBufferGetBytesPerRowOfPlane, CVPixelBufferUnlockBaseAddress};
-
-pub struct YUVFrame {
```
Comment (Pranav2612000) [inline src/frame/mod.rs:19]: Is there a reason for moving this from here to `lib.rs` ?

Comment (NiiightmareXD) [reply src/frame/mod.rs:19]: Yea frame.rs is filled with macos only code but this struct is used in other places.

**Comment by clearlysid** [comment]
Thanks for the PR! I'll flick through this tomorrow ✅

**Comment by thisislvca** [comment]
Looking forward to it!

**Comment by thisislvca** [comment]
Btw this still doesn't work @NiiightmareXD:

**Comment by thisislvca** [comment]
Btw this still doesn't work @NiiightmareXD:

```
error: failed to parse manifest at `C:\Users\lucam\Downloads\scap-main\Cargo.toml`

Caused by:
  Dependency 'ac-ffmpeg' has different source paths depending on the build target. Each dependency must have a single canonical source path irrespective of build target.
```

**Comment by clearlysid** [comment]
@thisislvca you need to provide it your own FFmpeg binaries. @ZohebMOPO can you document exact steps here to help?

**Comment by thisislvca** [comment]
@clearlysid I did provide the binaries, the error is a standard Cargo.toml error when you have deps imported with different source paths. At least that's what I found online :)

**Comment by clearlysid** [comment]
Ah! to temporarily get around this for Windows, you can comment out the ac-ffmpeg import for mac (or change it to be the same as windows)

**Comment by clearlysid** [comment]
Merging this to `main`, but just a heads up to everyone: we will be deprecating the `main` and using `v0.0.1/main` as the primary one going forward. We did use a lot of @NiiightmareXD's code as reference to update stuff so this PR was very helpful!

**Comment by thisislvca** [comment]
Ok cool to know!

**Comment by NiiightmareXD** [comment]
> Btw this still doesn't work @NiiightmareXD:
> 
> ```
> error: failed to parse manifest at `C:\Users\lucam\Downloads\scap-main\Cargo.toml`
> 
> Caused by:
>   Dependency 'ac-ffmpeg' has different source paths depending on the build target. Each dependency must have a single canonical source path irrespective of build target.
> ```

Comment out the version thats for macos I did this because I don't think its good idea to down grade the ffmpeg library for macos when the problem is with windows.

**Comment by Pranav2612000** [comment]
@allcontributors please add @NiiightmareXD for code

**Pull request merged**